src/Reflector/Generic.php

```diff
@@ -34,10 +35,20 @@ public function __construct(SourceLocator $sourceLocator)
      */
     public function reflect(Identifier $identifier)
     {
-        return $this->reflectFromLocatedSource(
-            $identifier,
-            $this->sourceLocator->__invoke($identifier)
-        );
+        $aggregate = $this->sourceLocator;
+        if (!($aggregate instanceof AggregateSourceLocator)) {
+            $aggregate = new AggregateSourceLocator([$this->sourceLocator]);
+        }
+
+        foreach ($aggregate->__invoke($identifier) as $locatedSource) {
+            try {
+                return $this->reflectFromLocatedSource($identifier, $locatedSource);
+            }
+            catch (Exception\IdentifierNotFound $identifierNotFound) {
+            }
+        }
+
+        throw Exception\IdentifierNotFound::fromIdentifier($identifier);
```

Comment:
Should probably encapsulate the previous exceptions, in order to show tested paths. Overkill?